src/git_provider/mod.rs

```diff
@@ -1,7 +1,61 @@
 #![allow(async_fn_in_trait)]
 
+use core::fmt;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use color_eyre::Section;
+use eyre::Result;
+
 pub mod github;
 
+#[derive(EnumIter)]
+pub enum SupportedProviders {
+    GitHub,
+}
+
+impl fmt::Display for SupportedProviders {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            SupportedProviders::GitHub => write!(f, "GitHub"),
+        }
+    }
+}
+
+impl SupportedProviders {
+    pub fn get_providers() -> String {
+        let mut s = String::new();
+
+        for provider in SupportedProviders::iter() {
+            s.push_str(&format!("{}, ", provider));
+        }
+
+        s[..s.len() - 2].to_string()
```

Comment:
Fishy, but I'm ok with it 😂

Reply:
yeah it's not the prettiest, maybe I'll come up with something better. Anyway, at least it can't generate any errors